flux-kube/plugins/flux-kube.c

```diff
@@ -0,0 +1,324 @@
+                             FLUX_NODEID_ANY, 0,
+                             "{s:s}", "jobspec", jobspec))) { 
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to send RPC: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+    
+    if (flux_rpc_get_unpack (f, "{s:i}", "retval", &retval) != 0) {
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to "
+                                     "unpack RPC: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+
+    if (retval != 0) {
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube.py: %s: submit callback "
+                                     "failed: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+
+    if (alloc_start (p, id, R) < 0)
+        flux_jobtap_raise_exception (p, id, "alloc", 0,
+                                     "failed to commit R to kvs");
+
+    if (flux_jobtap_job_set_flag (p,
+                                  FLUX_JOBTAP_CURRENT_JOB,
+                                  "alloc-bypass") < 0)
+        return flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                            "alloc", 0,
+                                            "Failed to set alloc-bypass: %s",
+                                            strerror (errno));
+    return 0;
+}
+
+static int cleanup_cb (flux_plugin_t *p,
+                       const char *topic,
+                       flux_plugin_arg_t *args,
+                       void *arg)
+{
+    const char *jobspec = NULL;
+    flux_future_t *f;
+    int retval = -1;
+
+    if (!(jobspec = flux_jobtap_job_aux_get (p,
+                                       FLUX_JOBTAP_CURRENT_JOB,
+                                       "flux-kube::jobspec")))
+        return 0;
+
+    if (!(f = flux_rpc_pack (flux_jobtap_get_flux (p), "fluxkube.cancel", 
+                             FLUX_NODEID_ANY, 0,
+                             "{s:s}", "jobspec", jobspec))) { 
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB,
+                                     "alloc", 0,
+                                     "flux-kube: %s: Failed to send RPC: %s",
+                                     topic,
+                                     strerror (errno));
+        return -1;
+    }
+    
+    if (flux_rpc_get_unpack (f, "{s:i}", "retval", &retval) != 0) {
```

Comment:
Another synchronous get that should be replaced with a `flux_future_then(3)` callback.